R/Rules-methods.R

```diff
@@ -284,6 +284,219 @@ setMethod(
   }
 )
 
+## NextBestNCRMLoss ----
+
+#' @describeIn nextBest find the next best dose based on the NCRM method and
+#'   loss function.
+#'
+#' @aliases nextBest-NextBestNCRMLoss
+#'
+#' @export
+#' @example examples/Rules-method-nextBest-NextBestNCRMLoss.R
+#'
+setMethod("nextBest",
+  signature = signature(
+    nextBest = "NextBestNCRMLoss",
+    doselimit = "numeric",
+    samples = "Samples",
+    model = "Model",
+    data = "Data"
+  ),
+  definition = function(nextBest, doselimit, samples, model, data, ...) {
+    doselimit <- ifelse(missing(doselimit) || length(doselimit) == 0, Inf, doselimit)
+
+    # Matrix with samples from the dose-tox curve at the dose grid points.
+    samples_prob <- sapply(data@doseGrid, prob, model = model, samples = samples)
+
+    prob_underdosing <- colMeans(samples_prob < nextBest@target[1])
+    prob_target <- colMeans(
+      (samples_prob >= nextBest@target[1]) & (samples_prob <= nextBest@target[2])
+    )
+    prob_overdose <- colMeans(
+      (samples_prob > nextBest@overdose[1]) & (samples_prob <= nextBest@overdose[2])
```

Comment:
But the `nextBest@overdose`, `nextBest@target` are never set when creating `nextBest` class object with `NextBestNCRMLoss()` user constructor. Something is wrong here.